packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -96,8 +108,17 @@ export class BiconomySmartAccountV2 extends BaseSmartAccount {
 
     instance.implementationAddress = biconomySmartAccountConfig.implementationAddress ?? BICONOMY_IMPLEMENTATION_ADDRESSES_BY_VERSION.V2_0_0;
 
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    instance.defaultValidationModule = biconomySmartAccountConfig.defaultValidationModule;
+    if (biconomySmartAccountConfig.defaultValidationModule) {
+      instance.defaultValidationModule = biconomySmartAccountConfig.defaultValidationModule;
+    } else if (biconomySmartAccountConfig.module) {
```

Comment:
can you elaborate this step

instance.defaultValidationModule = await instance.checkAndCreateModule(biconomySmartAccountConfig);

Reply:
based on other discussion I would suggest for the time being we can remove this else if

Reply:
The "checkAndCreateModule()" method is responsible for creating the instance of the module based on what the user passes in as a parameter for "authorizationModuleType".